C_CPP/Data Structures/Linked List/XOR Linked List/README.md

```diff
@@ -0,0 +1,217 @@
+    }
+    free(current);
+    current = NULL;
+    cout << " > " << val << " deleted from the linked list.\n";
+    return head;
+}
+
+// Traverse the linked list from start to end
+void traverse(Node* head) {
+    cout << " > Traversed linked list: ";
+    Node* current = head;
+    Node* previous = NULL;
+    Node* next = NULL;
+    while(current != NULL) { // Traverse the linked list node by node
+        cout << current->data << " -> ";
+        next = XOR(previous, current->link);
+        previous = current;
+        current = next;
+    }
+    cout << "NULL\n\n";
+}
+
+// Reverse the linked list
+Node* reverse(Node* head) {
+    Node* current = head;
+    if(current == NULL) { // If linked list is empty
+        return NULL;
+    }
+    Node* previous = NULL;
+    Node* next = NULL;
+    while(XOR(previous, current->link) != NULL) { // Traverse till the end of the linked list
+        next = XOR(previous, current->link);
+        previous = current;
+        current = next;
+    }
+    head = current; // Update head to the last node of the linked list
+    cout << " > Current linked list reversed.\n";
+    return head;
+}
+
+int main() {
+    Node* head = NULL; // Head of the linked list
+    traverse(head);
+    head = insertNode(head, 10);
+    head = insertNode(head, 20);
+    head = insertNode(head, 30);
+    head = insertNode(head, 40);
+    head = insertNode(head, 50);
+    traverse(head);
+    head = reverse(head);
+    traverse(head);
+    head = deleteNode(head, 50);
+    traverse(head);
+    head = deleteNode(head, 20);
+    traverse(head);
+    head = deleteNode(head, 80);
+    traverse(head);
+    return 0;
+}
+```
+
+# Output
+```
+ > Traversed linked list: NULL
+
+ > 10 inserted in the linked list.
+ > 20 inserted in the linked list.
+ > 30 inserted in the linked list.
+ > 40 inserted in the linked list.
+ > 50 inserted in the linked list.
+ > Traversed linked list: 50 -> 40 -> 30 -> 20 -> 10 -> NULL
+
+ > Current linked list reversed.
+ > Traversed linked list: 10 -> 20 -> 30 -> 40 -> 50 -> NULL
+
+ > 50 deleted from the linked list.
+ > Traversed linked list: 10 -> 20 -> 30 -> 40 -> NULL
+
+ > 20 deleted from the linked list.
+ > Traversed linked list: 10 -> 30 -> 40 -> NULL
+
+ > 80 not present for deletion.
+ > Traversed linked list: 10 -> 30 -> 40 -> NULL
+```
+
+# Advantages of XOR Linked List
+- **Less** and efficient use of **memory**.
+- **Reversing** an XOR linked list is **easy** as we just need to point the HEAD to the last node.
+
+# Disadvantages of XOR Linked List
+- Little **increase** in code complexity.
+- **Not supported** by many other programming languages like **Java, JavaScript, Python, etc** due to absence of access to memory address of the object.
+- Traversal **not possible** if we need to start it from the **middle** of the XOR linked list as we do not have the **necessary information** to start.
+
+#### Contributor: [Keval Dhanani](https://github.com/OptimalLearner)
```

Comment:
@Saviour1001  Allowed?

Reply:
Removed the last line as I noticed no one is using it in their readme files in this domain. I added it as it was allowed in some other domain.